utils/merge_and_push_translations.sh

```diff
@@ -79,64 +147,36 @@ if [ -n "$(git status -s)" ]; then
   # Push source catalogs to Transifex
   tx push -s
   sleep 65 # wait for translation files to be updated
-fi
 
-# Pull All translations from Transifex
-# tx pull -a
-# Force-pull translations because some would get skipped accidentally
-tx pull -fa
-
-# Update authors file
-python3 utils/update_authors.py
-if [ $? -eq 0 ] ; then
-  echo "Updated authors";
-else
-  echo "Failed updating authors";
-  exit 1;
-fi
+  # Pull All translations from Transifex
+  # use force to make sure really all files get pulled
+  tx pull -a -f
 
-# Update appdata
-python3 utils/update_appdata.py
-if [ $? -eq 0 ] ; then
-  echo "Updated appdata";
-else
-  echo "Failed updating appdata";
-  exit 1;
+  # Undo one-liner diffs of pure timestamps with no other content
+  undo_oneliner_diffs
+  runUpdateScripts=2 # run, files have changed
 fi
 
-# Update statistics
-python3 utils/update_translation_stats.py
-if [ $? -eq 0 ] ; then
-  echo "Updated translation stats";
-else
-  echo "Failed to update translation stats";
-  exit 1;
+if [ -n "$runUpdateScripts" ]; then # if not run yet or pot file has changed
+  update_authors # in case something has changed unexpectedly
+  update_appdata # in case something has changed unexpectedly
+  update_statistics
```

Comment:
Yes, you should do these right after `tx pull` again, and leave it to git to see if there were changes.

But the functions could include checking the git status whether their source files have changed before running anything.

Reply:
I think this would not cover [case d) (from above)](#r1304847903)

Reply:
Yes.